src/Renci.SshNet/Sftp/SftpFileAttributes.cs

```diff
@@ -397,8 +435,8 @@ private SftpFileAttributes()
 
         internal SftpFileAttributes(DateTime lastAccessTime, DateTime lastWriteTime, long size, int userId, int groupId, uint permissions, IDictionary<string, string> extensions)
         {
-            LastAccessTime = _originalLastAccessTime = lastAccessTime;
-            LastWriteTime = _originalLastWriteTime = lastWriteTime;
+            LastAccessTimeUtc = _originalLastAccessTimeUtc = ToUniversalTime(lastAccessTime);
+            LastWriteTimeUtc = _originalLastWriteTimeUtc = ToUniversalTime(lastWriteTime);
```

Comment:
If both the *lastAccessTime* and *lastWriteTime* argument now hold UTC times (since you used `DateTime.FromFileTimeUtc` to construct them), then why convert to UTC again?

Reply:
It's [defensive programming](https://en.wikipedia.org/wiki/Defensive_programming) since I can't guarantee that I'll be passed DateTime values where Kind == UTC. The primary caller of the internal SftpFileAttributes constructor _does_ pass UTC times now, and in that case the ToUniversalTime method won't change the value.

However, several callers from the Renci.SshNet.Tests assembly (which is allowed to access that internal constructor via the InternalsVisibleToAttribute) pass in non-UTC DateTimes (e.g., DateTime.MinValue and DateTime.Now). Those cases have Kind equal to Unspecified and Local, respectively. Calling ToUniversalTime in the SftpFileAttributes constructor is required to handle those correctly. You can use Visual Studio's "Find All References" on the internal SftpFileAttributes constructor to see these cases.

So, the ToUniversalTime method is a safety check to ensure both the production cases and the unit test cases all convert times consistently to UTC.

Reply:
Please remove the `ToUniversalTime()`, and rename the arguments from **last\*Time** to **last\*TimeUtc**.
If necessary, the tests should be adapted.